src/hotspot/share/gc/shenandoah/shenandoahOldGeneration.cpp

```diff
@@ -181,7 +181,8 @@ const char* ShenandoahOldGeneration::name() const {
 }
 
 bool ShenandoahOldGeneration::contains(ShenandoahHeapRegion* region) const {
-  return region->affiliation() != YOUNG_GENERATION;
+  // TODO: Should this be region->is_old() instead?
+  return !region->is_young();
```

Comment:
Agree with proposed change.  How could this not be a problem?   Is it used?

Reply:
We have to be careful with this one because some uses expect to visit regions that are both old _and_ free (or young _and_ free). Maybe we should get rid of this method and be explicit about it?

Reply:
Yeah, I would have expected `ShenandoahOldGeneration::contains` to look like the alias for `is_old`. I think `ShenandoahYoungGeneration::contains` has the opposite problem. That is something for a separate PR.